R/uvhydrograph-fnc.R

```diff
@@ -1,133 +1,141 @@
+    
+    add_corrected_uv(uv2_pts, label=secondary_lbl, addToLegend=addToSecondaryLegend)
+    add_uncorrected_uv(uv2_pts_raw, label=secondary_lbl, addToLegend=addToSecondaryLegend)
+    
+    # gageHeight
+    if(isSeriesOfType(data, "secondarySeries", "Gage height")) {
+      add_stage_measurements(data, month=month, addToLegend=addToSecondaryLegend)
+      
+      shift_pts <- subsetByMonth(getUvHydro(data, "effectiveShifts"), month)
+      tertiary_lbl <- getUvLabel(data, "effectiveShifts")
+      shifted_pts <- uv2_pts
+      shifted_pts[['y']] <- shifted_pts[['y']] + shift_pts[['y']]
+      add_estimated_uv(shifted_pts, label=secondary_lbl, addToLegend=addToSecondaryLegend)
+      
+      #add effective shift axis, timeseries, and shift measurements
+      # NOTE: this is a third plot, so this has to come at the end of this method.
+      #third axis
+      measuredShifts <- subsetByMonth(getFieldVisitErrorBarsShifts(data), month)
+      add_third_axes(secondary_lims = secondary_lims, tertiary_pts = shift_pts, tertiary_lbl = tertiary_lbl, 
+                     measured_shift_pts = measuredShifts, addToLegend=addToSecondaryLegend)
+      
+      add_shift_measurements(measuredShifts, addToLegend=addToSecondaryLegend)
+    }
+    
+    add_uvhydro_axes(secondary_lims, ylab = secondary_lbl, ylog = FALSE)
+    addLegend(secondary_legend);
   }
 }
 
-add_edited_uv <- function(pts){
-  points(pts$x, pts$y, type = 'l', col = 'blue', lty = 4)
+subsetByMonth <- function(pts, onlyMonth) {
+  return(subset(pts, month == onlyMonth))
 }
 
-add_meas_w_error <- function(times, points){
-  
-  #baseline points
-}
-add_computed_uv <- function(pts){
-  
-  points(pts$x, pts$y, type = 'l', col = 'black', lty = 1)
-}
-
-add_estimated_uv <- function(pts){
-  
-  col = 'orange'
-  lty = 5
-  type = 'l'
+add_uncorrected_uv <- function(pts, label, addToLegend){
+  points(pts$x, pts$y, type = 'l', col = 'blue', lty = 4)
+  addToLegend(paste("Uncorrected UV ", label), NA, 'blue', 4)
 }
 
-add_approved_dv <- function(points){
-  pch = c(4, 15) # for x and box
-  col = 'blue'
-  type = 'p'
-  points(points$x, points$y, pch = pch[1], type = type, col = col, lwd = 2)
-  points(points$x, rep( 10 ^ par()$usr[3], length(points$y)), pch = pch[2], type = type, col = col)
+add_corrected_uv <- function(pts, label, addToLegend){
+  points(pts$x, pts$y, type = 'l', col = 'black', lty = 1)
+  addToLegend(paste("Corrected UV ", label), NA, "black", 1)
 }
 
-add_review_dv <- function(pts){
-  pch = c(4, 15) # for x and box
-  col = 'yellow'
-  type = 'p'
+add_estimated_uv <- function(pts, label, addToLegend){
+  points(pts$x, pts$y, type = 'l', col = 'orange', lty = 5)
+  addToLegend(paste("Estimated UV ", label), NA, "Orange", 5)
 }
 
-add_working_dv <- function(times, points){
-  pch = c(4, 15) # for x and box
-  col = 'yellow'
-  type = 'p'
+add_dv <- function(points, approvals, pch, label, addToLegend){
+  approvalColors = c("red", "yellow", "blue")
```

Comment:
personally, I prefer connecting these kinds of things explicitly, like 
```R
approvals <- list(
   'working' = list('name' = 'Working", 'color' = "red"),
   'review' = list('name' = "In-review", color = 'yellow'))
```
or even
```R
approvals = c("Working" = "red", "In-review"="yellow", "Approved"= "blue")
approvals[["In-review"]] # or
approvals[[2]]
names(approvals)[2]
```